src/Microsoft.AspNetCore.WebUtilities/Base64UrlTextEncoder.cs

```diff
@@ -26,17 +29,89 @@ public static string Encode(byte[] data)
         /// <returns>The decoded data.</returns>
         public static byte[] Decode(string text)
         {
-            return Convert.FromBase64String(Pad(text.Replace('-', '+').Replace('_', '/')));
+            return Convert.FromBase64String(DecodeToBase64String(text));
         }
 
-        private static string Pad(string text)
+        // To enable unit testing
+        internal static string EncodeInternal(string base64EncodedString)
         {
-            var padding = 3 - ((text.Length + 3) % 4);
-            if (padding == 0)
+            // trim '='
+            int endIndex;
+            for (endIndex = base64EncodedString.Length - 1; endIndex >= 0; endIndex--)
+            {
+                if (base64EncodedString[endIndex] != '=')
+                {
+                    break;
+                }
+            }
+
+            int finalStringLength = endIndex + 1;
+            if (finalStringLength == 0)
```

Comment:
This is actually fine (example: if the encoded string is `=` or `==`...probably this is not realistic but we have unit tests covering these scenarios)